app/src/lib/git/merge.ts

```diff
@@ -21,3 +25,42 @@ export async function getMergeBase(
   )
   return process.stdout.trim()
 }
+
+/**
+ * Generate the merge result from two branches in a repository
+ *
+ * @param repository The repository containing the branches to merge
+ * @param ours The current branch
+ * @param theirs Another branch to merge into the current branch
+ */
+export async function mergeTree(
+  repository: Repository,
+  ours: Branch,
+  theirs: Branch
+): Promise<MergeResult | null> {
+  console.time('getMergeBase')
```

Comment:
I'm going to clean these up after doing the UI work, but for now these are nice to help identify chronic cases with any of the operations underneath.